user/lib.rs

```diff
@@ -0,0 +1,19 @@
+use std::sync::Arc;
+
+use system::concepts::{Credential, PasswordHash, User};
+
+use crate::user_manager::UserManager;
+
+pub mod errors;
+pub mod user_manager;
+
+pub fn initialise_default_user(user_manager: &Arc<UserManager>) {
```

Comment:
You don't need `&Arc<>`, you can just use `&UserManager` here - I believe Arc<T> automatically is usable as `&T` (or sometimes you use `.as_ref()` when it gets confused I think)

Reply:
"Extract into method" feature in IDEA is not as good as expected turns out. Fixed in 16b0608fdba6a3ab35f078650ff0be39c34869d2.